### PR TITLE
Fix single instance behavior in debug mode

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

This PR fix the current "Single instance" behavior in Debug mode that conflicts with the Release instances of KeePassXC.

After this PR:
- You can run unlimited Debug-compiled instances
- If you are running a Release instances you can disable the "single instance" mode in the settings

This PR also revert the old patch that isn't required anymore and add the APACHE2 license that was missing in 2.2.0 (for the Base32 library copyright)

## Motivation and context

Using Debug instances along with Release one was pretty annoying

## How has this been tested?

Manual testing

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/core/Config.cpp [hunk 1]
```diff
@@ -107,6 +107,7 @@ void Config::init(const QString& fileName)
 {
     m_settings.reset(new QSettings(fileName, QSettings::IniFormat));
 
+    m_defaults.insert("SingleIstance", true);
```
Comment (phoerious) [inline src/core/Config.cpp]: You're missing an n here.

File: src/gui/Application.cpp [hunk 2]
```diff
@@ -100,6 +101,10 @@ Application::Application(int& argc, char** argv)
     if (!userName.isEmpty()) {
         identifier.append("-");
         identifier.append(userName);
+#ifdef QT_DEBUG
+        // In DEBUG mode don't interfere with Release istances
```
Comment (phoerious) [inline src/gui/Application.cpp]: Here too. :-D

File: src/gui/SettingsWidgetGeneral.ui [hunk 3]
```diff
@@ -34,6 +34,16 @@
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
+        <widget class="QCheckBox" name="singleIstanceCheckBox">
+         <property name="text">
+          <string>Run only a single window</string>
```
Comment (phoerious) [inline src/gui/SettingsWidgetGeneral.ui]: Maybe rephrase to "Start only a single instance of KeePassXC" ?

**Comment by TheZ3ro** [comment]
@phoerious rebased, should now be fixed
